simulator/infra/macro.h

```diff
@@ -286,4 +287,32 @@ static inline uint128 gen_reverse( uint128 /* src1 */, size_t /* shamt */) {
     throw std::runtime_error( "Generalized reverse is not implemented for RV128");
     return 0;
 }
+
+static inline uint32 gen_or_combine( uint32 src1, size_t shamt)
+{
+    static std::array<uint32, 10> masks = { 0x5555'5555, 0x3333'3333, 0x0F0F'0F0F, 0x00FF'00FF, 0x0000'FFFF };
+    for(std::size_t i = 1, j = 0; i <= 16; i *= 2, j++)
+        if (shamt &  i)
```

Comment:
Can we define  `auto shift = (1 << j)` instead of double accounting of `i` and `j`?